### DISCUSSION TITLE
Standard & Advanced Playlists together

### DISCUSSION THREAD

**Discussion by DJMic** [AzuraCast/AzuraCast#7146] [Q&A]
Is it possible to use some playlists configured as "Standard" and some playlists configured as "Advanced" ?
I would like to use Advanced Playlists while keeping the advantages of AutoDJ.

Other question, when using Advanced Playlist, is option "Avoid Dupplicate Artists/Titles" working ?

**Comment by gAlleb** [comment]
Hi @DJMic,

> Is it possible to use some playlists configured as "Standard" and some playlists configured as "Advanced" ?
I would like to use Advanced Playlists while keeping the advantages of AutoDJ.

Yes, it is totally possible.

> Other question, when using Advanced Playlist, is option "Avoid Dupplicate Artists/Titles" working ?

"Avoid Dupplicate Artists/Titles" only works with non-advanced playlists.

<hr>
However, since your playlist is `advanced` - you can apply your own `Avoid Dupplicate` function (put it into you Liquidsoap configuration) .
Say, you want to avoid same artist duplicate within 2 hour interval:

```ruby
# Redefine your playlist again and add "check_next=check"

# Add some words for logging to be more precise. Sometime we have same artist but with different name: 
# e.g. Siouxsie and the Banshees / Siouxsie & the Banshees

stopwords=[
"a",
"with",
",",
"&",
"the",
"feat",
"vs",
"and"
]

# Sort the words from artist name

def tokenize(a)
  b = string.case(a, lower=true)
  c = r/[^\w'-]+/.split(b)
  d = list.filter(fun(i) -> not list.mem(i, stopwords), c)
  list.sort.natural(d)
end
 
# Define metadata to log

def my_hash(m)
  artist = metadata.artist(m)
  string.concat(tokenize(artist), separator=" ")
end

# Define log file.  duration=10000. means log history will be saved for the last 10000 seconds only

l = playlog(persistency="/var/azuracast/stations/radio/media/check_artist_radio.log", duration=10000.,hash=my_hash)


# Check the log file for duplicates. If an artist has played less then 7200 seconds ago - it would be rejected

def check(r)
    m = request.metadata(r)
   #
   # Specify time in seconds for avoiding same artist
   #
    if l.last(m) < 7200. then
    log(level=3,"Rejecting #{m['artist']} (played #{l.last(m)}s ago).")
      false
    else
      l.add(m)
      true
    end
end

# Add check_next=check to your playlist operator 

playlist = playlist(check_next=check, id="playlist", mime_type="audio/x-mpegurl",mode="randomize",reload_mode="watch","/var/azuracast/stations/radio/playlists/playlist.m3u")

**Reply by DJMic** [reply]
Great. Thanks for your answer.
I suppose "Manual Auto DJ Mode" must stay deactivated to use Standard and Advanced Playlists.

**Reply by gAlleb** [reply]
Yes.

**Reply by kikou2022** [reply]
Hi ! I've got an issue about that job [here  ](https://github.com/AzuraCast/AzuraCast/discussions/8181)
